tests/integration/ui_write/test_modular_storage.py

```diff
@@ -78,6 +79,23 @@ def roundtripContainer(self):
         with HDF5IO(self.link_filename, 'r', manager=get_manager()) as self.link_file_reader:
             return self.getContainer(self.link_file_reader.read())
 
+    def validate(self):
+        filenames = [self.data_filename, self.link_filename]
+        for fn in filenames:
+            if os.path.exists(fn):
+                with NWBHDF5IO(fn, mode='r') as io:
+                    errors = pynwb_validate(io)
+                    if errors:
+                        for err in errors:
+                            raise Exception(err)
+
+                # necessary to remove all references to the file and garbage
+                # collect on windows in order to be able to truncate/overwrite
+                # the file later. see pynwb GH issue #975
+                if os.name == 'nt':
+                    del io
+                    gc.collect()
+
     def getContainer(self, nwbfile):
```

Comment:
This test creates two linked files and so validation needed to be handled specially.